src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackDecoder.cs

```diff
@@ -0,0 +1,525 @@
+        private const byte IndexedHeaderStaticMask = 0x40;
+        private const byte IndexedHeaderStaticRepresentation = 0x40;
+        private const byte IndexedHeaderFieldPrefixMask = 0x3F;
+        private const int IndexedHeaderFieldPrefix = 6;
+
+        //0   1   2   3   4   5   6   7
+        //+---+---+---+---+---+---+---+---+
+        //| 0 | 0 | 0 | 1 |  Index(4+)   |
+        //+---+---+---+---+---------------+
+        private const byte PostBaseIndexMask = 0xF0;
+        private const int PostBaseIndexPrefix = 4;
+
+        //0   1   2   3   4   5   6   7
+        //+---+---+---+---+---+---+---+---+
+        //| 0 | 1 | N | S |Name Index(4+)|
+        //+---+---+---+---+---------------+
+        //| H |     Value Length(7+)     |
+        //+---+---------------------------+
+        //|  Value String(Length bytes)  |
+        //+-------------------------------+
+        private const byte LiteralHeaderFieldStaticMask = 0x10;
+        private const byte LiteralHeaderFieldPrefixMask = 0x0F;
+        private const int LiteralHeaderFieldPrefix = 4;
+
+        //0   1   2   3   4   5   6   7
+        //+---+---+---+---+---+---+---+---+
+        //| 0 | 0 | 0 | 0 | N |NameIdx(3+)|
+        //+---+---+---+---+---+-----------+
+        //| H |     Value Length(7+)     |
+        //+---+---------------------------+
+        //|  Value String(Length bytes)  |
+        //+-------------------------------+
+        private const byte LiteralHeaderFieldPostBasePrefixMask = 0x07;
+        private const int LiteralHeaderFieldPostBasePrefix = 3;
+
+        //0   1   2   3   4   5   6   7
+        //+---+---+---+---+---+---+---+---+
+        //| 0 | 0 | 1 | N | H |NameLen(3+)|
+        //+---+---+---+---+---+-----------+
+        //|  Name String(Length bytes)   |
+        //+---+---------------------------+
+        //| H |     Value Length(7+)     |
+        //+---+---------------------------+
+        //|  Value String(Length bytes)  |
+        //+-------------------------------+
+        private const byte LiteralHeaderFieldWithoutNameReferenceHuffmanMask = 0x08;
+        private const byte LiteralHeaderFieldWithoutNameReferencePrefixMask = 0x07;
+        private const int LiteralHeaderFieldWithoutNameReferencePrefix = 3;
+
+        private const int StringLengthPrefix = 7;
+        private const byte HuffmanMask = 0x80;
+
+        private const int DefaultStringBufferSize = 64;
+
+        private readonly int _maxHeadersLength;
+        private readonly DynamicTable _dynamicTable;
+        private State _state = State.Ready;
+
+        private byte[] _stringOctets;
+        private byte[] _headerNameOctets;
+        private byte[] _headerValueOctets;
+
+        // TODO idk what these are for.
+        private int _requiredInsertCount;
+        //private int _insertCount;
+        private int _base;
+
+        // s is used for whatever s is in each field. This has multiple definition
+        private bool _s;
```

Comment:
Is "s" actually the named used in the HTTP3 spec?  If yes, ok.  If no, can we pick something better? :smile:

Reply:
It was when @jkotalik wrote this against QPACK draft 10. Now in draft 11 it's called "t".

Regardless, I agree it can be changed.

Reply:
I tried to keep it as literal to the spec as possible. Guess I got bamboozled though and need to rename it to 't'. I think the spec has been changed to make T always represent whether it is in the static or dynamic table. https://quicwg.org/base-drafts/draft-ietf-quic-qpack.html#name-header-block-representation